marketplaceItems/keyboardBlueprint/entityClientScripts/keyboardZone.js

```diff
@@ -0,0 +1,126 @@
+//
+//  keyboardZone.js
+//
+//  created by Rebecca Stankus on 01/16/18
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* global Entities, MyAvatar, Script */
+
+(function() {
+    var MIN_DISTANCE_TO_FINGERTIP = 0.001;
+    var FINGER_ENTITY_DIMENSIONS = {x: 0.005,y: 0.005,z: 0.005};
+    var POSITION_CHECK_INTERVAL_MS = 100;
+    var POSITION_CHECK_TIMEOUT_MS = 3000;
+
+    var fingerEntities = [];
+    var interval;
+    var _this;
+
+    var Keyboard = function() {
+        _this = this;
+    };
+
+    Keyboard.prototype = {
+
+        /* ON PRELOAD: Save a reference to this key */
+        preload: function(entityID) {
+            _this.entityID = entityID;
+        },
+
+        /* ENTER ENTITY: Upon entering this zone around the keyboard, create small sphere entities at the tip of each 
+        finger and set up an interval that checks the distance between each sphere entity and its respective fingertip 
+        every 100 MS for 3 seconds. This interval will account for cases where the entities were not placed in the 
+        correct position, usually due avatar movement upon entering the zone, by moving the sphere closer to the 
+        correct position on every interval check if possible. The keyboard base is found and hand touch is removed 
+        for it and all of its child entities as all parts of the keyboard are parented to this base. This will 
+        prevent fingers from moving uncontrollably upon approaching parts of the keyboard. */
+        enterEntity: function() {
+            _this.createFingertipEntity("LeftHandIndex4");
+            _this.createFingertipEntity("LeftHandThumb4");
+            _this.createFingertipEntity("LeftHandPinky4");
+            _this.createFingertipEntity("LeftHandRing4");
+            _this.createFingertipEntity("LeftHandMiddle4");
+            _this.createFingertipEntity("RightHandIndex4");
+            _this.createFingertipEntity("RightHandThumb4");
+            _this.createFingertipEntity("RightHandPinky4");
+            _this.createFingertipEntity("RightHandRing4");
+            _this.createFingertipEntity("RightHandMiddle4");
+            interval = Script.setInterval(_this.updatePositions, POSITION_CHECK_INTERVAL_MS);
+            Script.setTimeout(function () {
+                Script.clearInterval(interval);
```

Comment:
Make sure to check if `interval` is running in here and only clear it if it is. Also in that case, make sure to set `interval` to `false`.